libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1718,6 +1718,19 @@ bool ExpressionCompiler::visit(MemberAccess const& _memberAccess)
 			solAssert(false, "Illegal fixed bytes member.");
 		break;
 	}
+	case Type::Category::Module:
+	{
+		Type::Category category = _memberAccess.annotation().type->category();
+		solAssert(
+			category == Type::Category::TypeType ||
+			category == Type::Category::Enum ||
+			category == Type::Category::Struct ||
+			category == Type::Category::Contract ||
```

Comment:
Are these tree actually possible? This would mean that the module contains an instance and I think that contract/enum/struct instances aren't allowed at module level.

If it's somehow possible then I think it would be a good idea to add a test case for this.